server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -712,21 +708,35 @@ public void run() {
      * migration will refer to a non-existing tablet, so it can never complete. Periodically scan
      * the metadata table and remove any migrating tablets that no longer exist.
      */
-    private void cleanupNonexistentMigrations(final AccumuloClient accumuloClient)
+    private void cleanupNonexistentMigrations(final ClientContext clientContext)
         throws TableNotFoundException {
-      Scanner scanner = accumuloClient.createScanner(MetadataTable.NAME, Authorizations.EMPTY);
-      TabletColumnFamily.PREV_ROW_COLUMN.fetch(scanner);
-      scanner.setRange(MetadataSchema.TabletsSection.getRange());
-      Set<KeyExtent> notSeen;
+
+      Map<DataLevel,Set<KeyExtent>> notSeen = new EnumMap<>(DataLevel.class);
+
       synchronized (migrations) {
-        notSeen = new HashSet<>(migrations.keySet());
+        // partition migrations by level
+        migrations.keySet().forEach(extent -> notSeen
```

Comment:
A `partitionMigrations` method was merged in yesterday.

Reply:
I used that method, had to move it because it was in an inner class.